### ISSUE TITLE
Verifying Deployment: Failed

### ISSUE THREAD

**Issue by ktquiet** [zodern/meteor-up#648]
Hello! I am trying to move a meteor app that is running successfully on Galaxy over to Digital Ocean to save money. I tried the Meteor Test app using the Ubuntu 14.04.5 x64 structure and I was successful using the Nginx server instructions.

Now I am trying to deploy my code (successful on Galaxy) over to the same droplet. It was hanging at the "Start Meteor" step. So I researched the issue in this file and saw the suggestion of making a new droplet AND uninstalling and reinstalling MUP.

I did that. I made a new Ubuntu 16.04.2 x64 droplet (after seeing others have more success with that) and it now clears the "Start Meteor" step. Where I am getting the error is in the "Verifying Deployment" step. I tried to increase the wait time from 60 to 90 seconds in mup.json but that did not work and I am stuck!

Below are my logs from all that I have explained above. you can see where I tried different things according to what I researched. I'm a new programmer so I could really use your help!

Katies-iMac:meteorfullversion katiecue$ mup setup

Started TaskList: Setup Docker
[138.68.44.2] - Setup Docker
[138.68.44.2] - Setup Docker: SUCCESS

Started TaskList: Setup Meteor
[138.68.44.2] - Setup Environment
[138.68.44.2] - Setup Environment: SUCCESS

Started TaskList: Setup Mongo
[138.68.44.2] - Setup Environment
[138.68.44.2] - Setup Environment: SUCCESS
[138.68.44.2] - Copying mongodb.conf
[138.68.44.2] - Copying mongodb.conf: SUCCESS

Started TaskList: Start Mongo
[138.68.44.2] - Start Mongo
[138.68.44.2] - Start Mongo: SUCCESS

Next, you should run:
    mup deploy
Katies-iMac:meteorfullversion katiecue$ mup deploy
Building App Bundle Locally

Started TaskList: Pushing Meteor App
[138.68.44.2] - Pushing Meteor App Bundle to The Server
[138.68.44.2] - Pushing Meteor App Bundle to The Server: SUCCESS

Started TaskList: Configuring App
[138.68.44.2] - Pushing the Startup Script
[138.68.44.2] - Pushing the Startup Script: SUCCESS
[138.68.44.2] - Sending Environment Variables
[138.68.44.2] - Sending Environment Variables: SUCCESS

Started TaskList: Start Meteor
[138.68.44.2] - Start Meteor
[138.68.44.2] x Start Meteor: FAILED
	
	-----------------------------------STDERR-----------------------------------
	Removing docker containers. Errors about nonexistent endpoints and containers are normal.
	Error response from daemon: endpoint meteorfullversion not found
	Error response from daemon: No such container: meteorfullversion-frontend
	Error response from daemon: endpoint meteorfullversion-frontend not found
	Error response from daemon: No such container: meteorfullversion-nginx-letsencrypt
	Error response from daemon: endpoint meteorfullversion-nginx-letsencrypt not found
	Error response from daemon: No such container: meteorfullversion-nginx-proxy
	Error response from daemon: endpoint meteorfullversion-nginx-proxy not found
	Finished removing docker containers
	docker: Error response from daemon: driver failed programming external connectivity on endpoint meteorfullversion (5a3226097722b28bfc64ff1bec83dee1ad33de55be36ad67b0d73a6aaa89c910): Bind for 0.0.0.0:80 failed: port is already allocated.
	-----------------------------------STDOUT-----------------------------------
	meteorfullversion
	base: Pulling from abernix/meteord
	Digest: sha256:9ed85313621010dd27a08054ae9225aa1f31798647fbcfdc9cc082181c6f9c97
	Status: Image is up to date for abernix/meteord:base
	Pulled abernix/meteord:base
	485b9f78ef99af1e77011ab50d73a7641059b459e77b096e256d5bea19c81741
	----------------------------------------------------------------------------
Katies-iMac:meteorfullversion katiecue$ node -v
v7.8.0
Katies-iMac:meteorfullversion katiecue$ npm install -g mup
npm WARN checkPermissions Missing write access to /usr/local/lib/node_modules/mup/node_modules
npm ERR! path /usr/local/lib/node_modules/mup/node_modules
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall access
npm ERR! Error: EACCES: permission denied, access '/usr/local/lib/node_modules/mup/node_modules'
npm ERR!  { Error: EACCES: permission denied, access '/usr/local/lib/node_modules/mup/node_modules'
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/usr/local/lib/node_modules/mup/node_modules' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.

┌─────────────────────────────────────────────────────────────┐
│                   npm update check failed                   │
│             Try running with sudo or get access             │
│            to the local update config store via             │
│ sudo chown -R $USER:$(id -gn $USER) /Users/katiecue/.config │
└─────────────────────────────────────────────────────────────┘

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/katiecue/.npm/_logs/2017-06-20T19_06_22_469Z-debug.log
Katies-iMac:meteorfullversion katiecue$ sudo npm install -g mup
Password:
/usr/local/bin/mup -> /usr/local/lib/node_modules/mup/index.js
+ mup@1.2.11
added 1 package and updated 3 packages in 5.219s
Katies-iMac:meteorfullversion katiecue$ mup init
Skipping creation of settings.json.
settings.json already exist at /Users/katiecue/meteorfullversion/settings.json.
Skipping creation of mup.js
mup.js already exists at /Users/katiecue/meteorfullversion/mup.js
Katies-iMac:meteorfullversion katiecue$ mup setup

Started TaskList: Setup Docker
[138.68.44.2] - Setup Docker
[138.68.44.2] - Setup Docker: SUCCESS

Started TaskList: Setup Meteor
[138.68.44.2] - Setup Environment
[138.68.44.2] - Setup Environment: SUCCESS

Started TaskList: Setup Mongo
[138.68.44.2] - Setup Environment
[138.68.44.2] - Setup Environment: SUCCESS
[138.68.44.2] - Copying mongodb.conf
[138.68.44.2] - Copying mongodb.conf: SUCCESS

Started TaskList: Start Mongo
[138.68.44.2] - Start Mongo
[138.68.44.2] - Start Mongo: SUCCESS

Next, you should run:
    mup deploy
Katies-iMac:meteorfullversion katiecue$ mup deploy
Building App Bundle Locally

Started TaskList: Pushing Meteor App
[138.68.44.2] - Pushing Meteor App Bundle to The Server
[138.68.44.2] - Pushing Meteor App Bundle to The Server: SUCCESS

Started TaskList: Configuring App
[138.68.44.2] - Pushing the Startup Script
[138.68.44.2] - Pushing the Startup Script: SUCCESS
[138.68.44.2] - Sending Environment Variables
[138.68.44.2] - Sending Environment Variables: SUCCESS

Started TaskList: Start Meteor
[138.68.44.2] - Start Meteor
[138.68.44.2] x Start Meteor: FAILED
	
	-----------------------------------STDERR-----------------------------------
	Removing docker containers. Errors about nonexistent endpoints and containers are normal.
	Error response from daemon: endpoint meteorfullversion not found
	Error response from daemon: No such container: meteorfullversion-frontend
	Error response from daemon: endpoint meteorfullversion-frontend not found
	Error response from daemon: No such container: meteorfullversion-nginx-letsencrypt
	Error response from daemon: endpoint meteorfullversion-nginx-letsencrypt not found
	Error response from daemon: No such container: meteorfullversion-nginx-proxy
	Error response from daemon: endpoint meteorfullversion-nginx-proxy not found
	Finished removing docker containers
	docker: Error response from daemon: driver failed programming external connectivity on endpoint meteorfullversion (71cb31d6b0451afc821933d0db3049098296f432153c98ff020cdfe73be03bd5): Bind for 0.0.0.0:80 failed: port is already allocated.
	-----------------------------------STDOUT-----------------------------------
	meteorfullversion
	base: Pulling from abernix/meteord
	Digest: sha256:9ed85313621010dd27a08054ae9225aa1f31798647fbcfdc9cc082181c6f9c97
	Status: Image is up to date for abernix/meteord:base
	Pulled abernix/meteord:base
	d8bc013b03198715b12aa7d601a02d095e4bd7d08ba6f76bc360eb3679703010
	----------------------------------------------------------------------------
Katies-iMac:meteorfullversion katiecue$ sudo npm uninstall -g mup
Password:
removed 198 packages in 7.677s
Katies-iMac:meteorfullversion katiecue$ sudo npm install -g mup
/usr/local/bin/mup -> /usr/local/lib/node_modules/mup/index.js
+ mup@1.2.11
added 197 packages in 14.999s
Katies-iMac:meteorfullversion katiecue$ mup init
Skipping creation of settings.json.
settings.json already exist at /Users/katiecue/meteorfullversion/settings.json.
Skipping creation of mup.js
mup.js already exists at /Users/katiecue/meteorfullversion/mup.js
Katies-iMac:meteorfullversion katiecue$ mup setup

Started TaskList: Setup Docker
[138.68.253.99] - Setup Docker
[138.68.253.99] - Setup Docker: SUCCESS

Started TaskList: Setup Meteor
[138.68.253.99] - Setup Environment
[138.68.253.99] - Setup Environment: SUCCESS

Started TaskList: Setup Mongo
[138.68.253.99] - Setup Environment
[138.68.253.99] - Setup Environment: SUCCESS
[138.68.253.99] - Copying mongodb.conf
[138.68.253.99] - Copying mongodb.conf: SUCCESS

Started TaskList: Start Mongo
[138.68.253.99] - Start Mongo
[138.68.253.99] - Start Mongo: SUCCESS

Next, you should run:
    mup deploy
Katies-iMac:meteorfullversion katiecue$ mup deploy
Building App Bundle Locally

Started TaskList: Pushing Meteor App
[138.68.253.99] - Pushing Meteor App Bundle to The Server
[138.68.253.99] - Pushing Meteor App Bundle to The Server: SUCCESS

Started TaskList: Configuring App
[138.68.253.99] - Pushing the Startup Script
[138.68.253.99] - Pushing the Startup Script: SUCCESS
[138.68.253.99] - Sending Environment Variables
[138.68.253.99] - Sending Environment Variables: SUCCESS

Started TaskList: Start Meteor
[138.68.253.99] - Start Meteor
[138.68.253.99] - Start Meteor: SUCCESS
[138.68.253.99] - Verifying Deployment
[138.68.253.99] x Verifying Deployment: FAILED
	
	-----------------------------------STDERR-----------------------------------
	teor-dev-bundle@0.0.0 No license field.
	=> Starting meteor app on port:80
	Note: you are using a pure-JavaScript implementation of bcrypt.
	While this implementation will work correctly, it is known to be
	approximately three times slower than the native implementation.
	In order to use the native implementation instead, run
	
	  meteor npm install --save bcrypt
	
	in the root directory of your application.
	/bundle/bundle/programs/server/node_modules/fibers/future.js:280
							throw(ex);
							^
	
	ReferenceError: module is not defined
	    at app/mup.js:1:-27
	    at app/mup.js:68:4
	    at /bundle/bundle/programs/server/boot.js:295:34
	    at Array.forEach (native)
	    at Function._.each._.forEach (/bundle/bundle/programs/server/node_modules/underscore/underscore.js:79:11)
	    at /bundle/bundle/programs/server/boot.js:128:5
	    at /bundle/bundle/programs/server/boot.js:344:5
	    at Function.run (/bundle/bundle/programs/server/profile.js:480:12)
	    at /bundle/bundle/programs/server/boot.js:343:11
	-----------------------------------STDOUT-----------------------------------
	
	To see more logs type 'mup logs --tail=100'
	
	----------------------------------------------------------------------------
Katies-iMac:meteorfullversion katiecue$ meteor npm install --save bcrypt

> bcrypt@1.0.2 install /Users/katiecue/meteorfullversion/node_modules/bcrypt
> node-pre-gyp install --fallback-to-build

module.js:327
    throw err;
    ^

Error: Cannot find module '../'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/katiecue/meteorfullversion/node_modules/.bin/node-pre-gyp:15:20)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
- jodid25519@1.0.2 node_modules/jodid25519
npm ERR! Darwin 16.6.0
npm ERR! argv "/Users/katiecue/.meteor/packages/meteor-tool/.1.4.2_3.1k8h8ke++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/node" "/Users/katiecue/.meteor/packages/meteor-tool/.1.4.2_3.1k8h8ke++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/npm" "install" "--save" "bcrypt"
npm ERR! node v4.6.2
npm ERR! npm  v3.10.9
npm ERR! code ELIFECYCLE

npm ERR! bcrypt@1.0.2 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the bcrypt@1.0.2 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the bcrypt package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs bcrypt
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls bcrypt
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/katiecue/meteorfullversion/npm-debug.log
Katies-iMac:meteorfullversion katiecue$ mup setup

Started TaskList: Setup Docker
[138.68.253.99] - Setup Docker
[138.68.253.99] - Setup Docker: SUCCESS

Started TaskList: Setup Meteor
[138.68.253.99] - Setup Environment
[138.68.253.99] - Setup Environment: SUCCESS

Started TaskList: Setup Mongo
[138.68.253.99] - Setup Environment
[138.68.253.99] - Setup Environment: SUCCESS
[138.68.253.99] - Copying mongodb.conf
[138.68.253.99] - Copying mongodb.conf: SUCCESS

Started TaskList: Start Mongo
[138.68.253.99] - Start Mongo
[138.68.253.99] - Start Mongo: SUCCESS

Next, you should run:
    mup deploy
Katies-iMac:meteorfullversion katiecue$ mup deploy
Building App Bundle Locally

Started TaskList: Pushing Meteor App
[138.68.253.99] - Pushing Meteor App Bundle to The Server
[138.68.253.99] - Pushing Meteor App Bundle to The Server: SUCCESS

Started TaskList: Configuring App
[138.68.253.99] - Pushing the Startup Script
[138.68.253.99] - Pushing the Startup Script: SUCCESS
[138.68.253.99] - Sending Environment Variables
[138.68.253.99] - Sending Environment Variables: SUCCESS

Started TaskList: Start Meteor
[138.68.253.99] - Start Meteor
[138.68.253.99] - Start Meteor: SUCCESS
[138.68.253.99] - Verifying Deployment
[138.68.253.99] x Verifying Deployment: FAILED
	
	-----------------------------------STDERR-----------------------------------
	ting meteor app on port:80
	Note: you are using a pure-JavaScript implementation of bcrypt.
	While this implementation will work correctly, it is known to be
	approximately three times slower than the native implementation.
	In order to use the native implementation instead, run
	
	  meteor npm install --save bcrypt
	
	in the root directory of your application.
	/bundle/bundle/programs/server/node_modules/fibers/future.js:280
							throw(ex);
							^
	
	ReferenceError: module is not defined
	    at app/mup.js:1:-27
	    at app/mup.js:68:4
	    at /bundle/bundle/programs/server/boot.js:295:34
	    at Array.forEach (native)
	    at Function._.each._.forEach (/bundle/bundle/programs/server/node_modules/underscore/underscore.js:79:11)
	    at /bundle/bundle/programs/server/boot.js:128:5
	    at /bundle/bundle/programs/server/boot.js:344:5
	    at Function.run (/bundle/bundle/programs/server/profile.js:480:12)
	    at /bundle/bundle/programs/server/boot.js:343:11
	 
	=> Redeploying previous version of the app
	 
	-----------------------------------STDOUT-----------------------------------
	
	To see more logs type 'mup logs --tail=100'
	
	----------------------------------------------------------------------------
Katies-iMac:meteorfullversion katiecue$ sudo meteor npm install
Password:
inspinia@2.7.0 /Users/katiecue/meteorfullversion
└── (empty)

npm ERR! code 1
Katies-iMac:meteorfullversion katiecue$

**Comment by ktquiet** [comment]
Someone suggested I look at Meteor packages. I did

1. I removed and reinstalled accounts-password (caused trouble with Galaxy deployment too). Deployment still failed.
2. I updated packages. Deployment still failed.
3. I removed accounts-password . Deployment still failed but different error code. See bottom for newest error code 

Katies-iMac:meteorfullversion katiecue$ meteor list
accounts-password        1.3.3* Password support for accounts
autoupdate               1.3.12  Update the client when new client code is available
blaze-html-templates     1.0.5* Compile HTML templates into reactive UI with Meteor Blaze
check                    1.2.4* Check whether a value matches a pattern
dbarrett:dropzonejs      3.10.3* Dropzone.js - an open source library that provides drag'n'drop file uploads with image previews
ejson                    1.0.13  Extended and Extensible JSON library
fortawesome:fontawesome  4.7.0  Font Awesome (official): 500+ scalable vector icons, customizable via CSS, Retina friendly
jquery                   1.11.10  Manipulate the DOM using CSS selectors
kadira:blaze-layout      2.3.0  Layout Manager for Blaze (works well with FlowRouter)
kadira:flow-router       2.12.1  Carefully Designed Client Side Router for Meteor
less                     2.7.8* Leaner CSS language
logging                  1.1.16* Logging facility.
meteor-base              1.0.4* Packages that every Meteor app needs
mobile-experience        1.0.4  Packages for a great mobile user experience
mongo                    1.1.14* Adaptor for using MongoDB and Minimongo over DDP
random                   1.0.10  Random number generator and utilities
reload                   1.1.11  Reload the page while preserving application state.
session                  1.1.7  Session variable
shell-server             0.2.1* Server-side component of the `meteor shell` command.
spacebars                1.0.13* Handlebars-like template language for Meteor
standard-minifier-css    1.3.2* Standard css minifier used with Meteor apps by default.
standard-minifier-js     1.2.1* Standard javascript minifiers used with Meteor apps by default.
tracker                  1.1.1* Dependency tracker to allow reactive callbacks
twbs:bootstrap           3.3.6  The most popular front-end framework for developing responsive, mobile first projects on the web.
zimme:active-route       2.3.2  Active route helpers

                                              
* New versions of these packages are available! Run 'meteor update' to try to update those packages to their latest versions. If your packages cannot be updated further, try
  typing `meteor add <package>@<newVersion>` to see more information.
Katies-iMac:meteorfullversion katiecue$





------------------------



Started TaskList: Start Meteor
[138.68.253.99] - Start Meteor
[138.68.253.99] - Start Meteor: SUCCESS
[138.68.253.99] - Verifying Deployment
[138.68.253.99] x Verifying Deployment: FAILED
	
	-----------------------------------STDERR-----------------------------------
	DEV",
	  "http_parser": "2.7.0",
	  "icu": "56.1",
	  "modules": "46",
	  "node": "4.8.0",
	  "openssl": "1.0.2k",
	  "uv": "1.9.1",
	  "v8": "4.5.103.45",
	  "zlib": "1.2.8"
	}
	npm WARN meteor-dev-bundle@0.0.0 No description
	npm WARN meteor-dev-bundle@0.0.0 No repository field.
	npm WARN meteor-dev-bundle@0.0.0 No license field.
	=> Starting meteor app on port:80
	/bundle/bundle/programs/server/node_modules/fibers/future.js:280
							throw(ex);
							^
	
	ReferenceError: module is not defined
	    at app/mup.js:1:-27
	    at app/mup.js:68:4
	    at /bundle/bundle/programs/server/boot.js:338:34
	    at Array.forEach (native)
	    at Function._.each._.forEach (/bundle/bundle/programs/server/node_modules/underscore/underscore.js:79:11)
	    at /bundle/bundle/programs/server/boot.js:158:5
	    at /bundle/bundle/programs/server/boot.js:387:5
	    at Function.run (/bundle/bundle/programs/server/profile.js:510:12)
	    at /bundle/bundle/programs/server/boot.js:386:11
	 
	=> Redeploying previous version of the app
	 
	-----------------------------------STDOUT-----------------------------------
	
	To see more logs type 'mup logs --tail=100'
	
	----------------------------------------------------------------------------
